README.md
## Example with static load of gmaps API

### Install

yarn add react react-dom leaflet react-leaflet react-leaflet-googlemutant

### Inport with GMaps loader class

```js
import React, { Component } from 'react';
import { Map, LayersControl } from 'react-leaflet';
import { GoogleMutant, GoogleApiLoader } from 'react-leaflet-googlemutant';
import './App.css';

class App extends Component {
  render() {
    const { BaseLayer } = LayersControl;
    const mapConf = {
      zoom: 12,
      center: [-3.7925557,-38.497655],
      zoomControl: true,
      doubleClickZoom: false,
    };
    return (
      <div className="App">
        <GoogleApiLoader apiKey="AXSBVXSBVXSBVXBSVXS">
          <Map {...mapConf}>
            <LayersControl position='topright'>
              <BaseLayer checked name='Google Maps Roads'>
                <GoogleMutant type="roadmap"/>
              </BaseLayer>
              <BaseLayer name='Google Maps Terrain'>
                <GoogleMutant type="terrain" />
              </BaseLayer>
              <BaseLayer name='Google Maps Satellite'>
                <GoogleMutant type="satellite" />
              </BaseLayer>
              <BaseLayer  name='Google Maps Hydrid'>
                <GoogleMutant type="hybrid" />
              </BaseLayer>
            </LayersControl>
          </Map>
        </GoogleApiLoader>
      </div>
    );
  }
}

export default App;
```

### Import with script into html

```html
<script src="https://maps.google.com/maps/api/js?v=3.2" charset="utf-8"></script>
```

### Considering that there is a global variable as **google**

```js

import React, { Component } from 'react';
import { Map, LayersControl } from 'react-leaflet';
import GoogleMutant from 'react-leaflet-googlemutant';
import './App.css';

class App extends Component {
  render() {
    const { BaseLayer } = LayersControl;
    const mapConf = {
      zoom: 12,
      center: [-3.7925557,-38.497655],
      zoomControl: true,
      doubleClickZoom: false,
    };
    return (
      <div className="App">
        <Map {...mapConf}>
          <LayersControl position='topright'>
            <BaseLayer checked name='Google Maps Roads'>
              <GoogleMutant type="roadmap"/>
            </BaseLayer>
            <BaseLayer name='Google Maps Terrain'>
              <GoogleMutant type="terrain" />
            </BaseLayer>
            <BaseLayer name='Google Maps Satellite'>
              <GoogleMutant type="satellite" />
            </BaseLayer>
            <BaseLayer  name='Google Maps Hydrid'>
              <GoogleMutant type="HIBRID" />
            </BaseLayer>
          </LayersControl>
        </Map>
      </div>
    );
  }
}

export default App;

```

## License

MIT  
See [LICENSE](LICENSE) file.


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="0.1.20"></a>
## [0.1.20](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.19...v0.1.20) (2018-01-25)



<a name="0.1.19"></a>
## [0.1.19](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.18...v0.1.19) (2018-01-25)



<a name="0.1.18"></a>
## [0.1.18](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.17...v0.1.18) (2017-12-15)



<a name="0.1.17"></a>
## [0.1.17](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.16...v0.1.17) (2017-12-15)



<a name="0.1.16"></a>
## [0.1.16](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.15...v0.1.16) (2017-12-14)



<a name="0.1.15"></a>
## [0.1.15](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.14...v0.1.15) (2017-10-03)



<a name="0.1.13"></a>
## [0.1.13](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.12...v0.1.13) (2017-10-03)



<a name="0.1.12"></a>
## [0.1.12](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.11...v0.1.12) (2017-07-21)



<a name="0.1.11"></a>
## [0.1.11](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.10...v0.1.11) (2017-07-17)



<a name="0.1.10"></a>
## [0.1.10](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.9...v0.1.10) (2017-06-21)



<a name="0.1.9"></a>
## [0.1.9](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.8...v0.1.9) (2017-06-21)



<a name="0.1.8"></a>
## [0.1.8](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.7...v0.1.8) (2017-06-20)



<a name="0.1.7"></a>
## [0.1.7](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.6...v0.1.7) (2017-06-19)



<a name="0.1.6"></a>
## [0.1.6](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.5...v0.1.6) (2017-06-09)



<a name="0.1.5"></a>
## [0.1.5](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.4...v0.1.5) (2017-06-09)



<a name="0.1.4"></a>
## [0.1.4](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.3...v0.1.4) (2017-06-09)



<a name="0.1.3"></a>
## [0.1.3](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.2...v0.1.3) (2017-06-07)



<a name="0.1.2"></a>
## [0.1.2](https://github.com/produtoreativo/react-leaflet-googlemutant/compare/v0.1.1...v0.1.2) (2017-06-07)



<a name="0.1.1"></a>
## 0.1.1 (2017-06-07)


src/App.js
import React, { Component } from 'react';
import { Map, LayersControl } from 'react-leaflet';
import GoogleApiLoader from 'react-leaflet-googlemutant/googleapiloader';
import GoogleMutant from 'react-leaflet-googlemutant/googlemutant';
import './App.css';

class App extends Component {
  render() {
    const { BaseLayer } = LayersControl;
    const mapConf = {
      zoom: 12,
      center: [-3.7925557,-38.497655],
      zoomControl: true,
      doubleClickZoom: false,
    };
    return (
      <div className="App">
        <GoogleApiLoader>
          <Map {...mapConf}>
            <LayersControl position='topright'>
              <BaseLayer checked name='Google Maps Roads'>
                <GoogleMutant type="roadmap"/>
              </BaseLayer>
              <BaseLayer name='Google Maps Terrain'>
                <GoogleMutant type="terrain" />
              </BaseLayer>
              <BaseLayer name='Google Maps Satellite'>
                <GoogleMutant type="satellite" />
              </BaseLayer>
              <BaseLayer  name='Google Maps hybrid'>
                <GoogleMutant type="hybrid" />
              </BaseLayer>
            </LayersControl>
          </Map>
        </GoogleApiLoader>
      </div>
    );
  }
}

export default App;
